Reducing Redis heap memory overhead when running ticket registry cleaner

gagarski opened the pull request:
I've noticed some bursts in memory usage when using `RedisTicketRegistry` together with `DefaultTicketRegistryCleaner`. 

![mem](https://i.imgur.com/5oKLl1D.jpg)

After some research I've found out that `DefaultTicketRegistryCleaner` loads all the tickets from Redis to Java heap memory due to default implementation of `getTicketsStream()` in `RedisTicketRegistry` (which calls `getTickets()`).

This PR does two things:
 - defines proper `getTicketsStream` in `RedisTicketRegistry`
 - fetches Redis KEYS in a proper way as Redis documentation suggests (using SCAN instead of KEYS).

gagarski commented on support/cas-server-support-redis-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/RedisTicketRegistry.java:
```diff
@@ -110,6 +119,27 @@ public Ticket updateTicket(final Ticket ticket) {
+            return StreamSupport
+                    .stream(Spliterators.spliteratorUnknownSize(cursor, Spliterator.ORDERED), false)
+                    .map((key) -> (String) client.getKeySerializer().deserialize(key))
+                    .collect(Collectors.toList())
```
I'd personally prefer to make a proper stream without materializing it to `List`. But we should make this stream closeable to properly close the `Cursor`. If we do it we should wrap `getKeysStream` and `getKeys(Predicate)` to try-with-resources everywhere. I can do it if it is OK.

mmoayyed replied:
Sure, sounds OK to me.

Also, please make sure you're using the correct indentations. (Looks like you're using tabs in some places)

gagarski replied:
I did not find any tabs in my code.

gagarski commented:
I've implemented proper stream for Redis keys and added try-with-resources. Also I've replaced `getTickets().stream()` call to `getTicketsStream()` everywhere except tests. 

I've briefly looked into other `TicketRegistry` implementations and it seems like most of them are sufferring from this problem. I probably can't fix all of them so I'll probably just port these changes onto master branch after CI is passed.
